test/example/stack-utils.cpp

```diff
@@ -297,6 +297,54 @@ void test_signature_satisfaction() {
   }
 }
 
+void test_signature_lub() {
+  std::cout << ";; Test stack signature lub\n";
+  {
+    StackSignature a{Type::none, Type::none, false};
+    StackSignature b{Type::none, Type::none, false};
```

Comment:
separate from this PR, the bool at the end makes this harder to read I think. An enum of Reachable/Unreachable might be clearer?

Reply:
Agreed. I will do this as a follow-up.